### PR TITLE
Operations

### PR DESCRIPTION
**Pull request by bergundy**
- Add top level `Operation` concept
- Change Nexus headers to accept single string value as a least common denominator for protocols we may want to support in the future
- Rename `Handler` interface to `ServiceHandler` to avoid encouraging its use of `OperationHandler`
- Add an `OperationDirectoryHandler` - likely will be used in most applications instead of implementing a `ServiceHandler` directly
- Unify client and server options structs
- Add a `HandlerErrorType` instead of relying on HTTP statuses

TODO:
- Fix/add all docstrings
- Fix README
- Add examples for using the operation APIs
- Consider abstracting HTTP in `completion.go`

### CODE REVIEW

File: nexus/operation.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+func (d OperationDefinition[I, O]) Name() string {
+	return string(d)
+}
+
+// inferType implements Operation.
+func (OperationDefinition[I, O]) inferType(I, O) {} //nolint:unused
+
+var _ Operation[any, any] = OperationDefinition[any, any]("")
+
+type UntypedOperationHandler interface {
```
Comment (bergundy) [inline nexus/operation.go:32]:
```suggestion
type RegisterableOperationHandler interface {
```

File: nexus/operation.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	Operations []UntypedOperationHandler
+}
+
+type OperationDirectoryHandler struct {
+	UnimplementedServiceHandler
+
+	operations map[string]UntypedOperationHandler
+}
+
+func NewOperationDirectoryHandler(options OperationDirectoryHandlerOptions) (*OperationDirectoryHandler, error) {
```
Comment (bergundy) [inline nexus/operation.go:87]:
```
r := NewOperationRegistry()
r.Register(...operations)
NewOperationRegistryHandler(r, options)
```

```
r := NewOperationRegistry()
r.Register(...operations)
r.NewServiceHandler(options)
```

```
opts := OperationDirectoryHandlerOptions{}
append(opts.Operations, ...)
```

```
r := OperationRegistry{}
r.Register(...operations)
r.NewServiceHandler(options) (ServiceHandler, error)
```

Comment (bergundy) [reply nexus/operation.go:87]: We'll go with the last one

**Review by bergundy — COMMENTED** [review]
```
ServiceHandler -> Handler
OperationDirectoryHandler -> Registry.NewHandler()
OperationHandler -> Operation
NewOperationDefinition (or Reference)
```

**Review by cretz — APPROVED** [review]
This is good enough to get moving with (any Go API designs we can revisit later)

**Pull request merged**